### PR TITLE
Setup pre-commit autoformatting with Husky

### LINKED ISSUE #84: Set up Husky pre-commit hook to run formatting
Following [discussion](https://github.com/simple-icons/release-action/pull/24#issuecomment-756949234), this is currently configured in [simple-icons/release-action](https://github.com/simple-icons/release-action). Just we must make sure that permissions of `.husky/pre-commit` file are the same across different systems (see [this](https://github.com/simple-icons/release-action/pull/24#issuecomment-756943291)).

### PR DESCRIPTION
**Pull request by mondeja**
Closes #84

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+. "$(dirname $0)/_/husky.sh"
+
+git stash -q --keep-index
+
+# Add formatted files to commit
+files=$(npm run format | tail -n +5) || exit $?
+printf "$files" | while read line ; do
+  file="$(printf "$line" | cut -d' ' -f1)"
+  git add "$file"
+done;
```
Comment (ericcornelissen) [inline .husky/pre-commit:11]: My shell scripting skills aren't that good, why not just:

```suggestion
# Format files and include formatting in commit
npm run format
git add .
```

The stashing on line 4 ensures non-staged files will not be included in the commit.

Comment (ericcornelissen) [reply .husky/pre-commit]: @mondeja, I do think you need to stage the results of `npm run format` using e.g. `git add .`

Also, just out of curiosity, what is the advantage of having the ` || exit $?` suffixed? If it is to ensure the script fails if `npm run format` fails, I believe this is handled by the Husky script invoked at line 2, but might be worth testing this manually to be sure.

Comment (mondeja) [reply .husky/pre-commit]: > I believe this is handled by the Husky script invoked at line 2, but might be worth testing this manually to be sure.

Yes, I didn't know, removed :+1:
